### PR TITLE
ROX-14179: Adjust traits message

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

This PR prepares the traits message for later usage within the declarative configuration project.

As defined [within the design doc](https://docs.google.com/document/d/1UmdaEzu_wRXtSzzyBJu2hUK4JW8Og14fENAHJpaY8_g/edit#heading=h.h2swd5w93ebw), the traits message shall be re-used to identify the origin from which an object was created.

As an origin, we currently have the following three in mind:
- Imperative - the object was created imperatively using the API.
- Default - the object is a default object created with each installation of stackrox (e.g. default image integrations, default access scopes etc.).
- Declarative - the object was created declaratively.

This will later on help with applying specific rules to these objects.
We envision that declaratively created objects may not reference imperatively created objects, and are not allowed to be modified via API.

This is the first step in establishing the new field in the proto message.
Additional PRs will follow that will adjust all proto messages with the traits message.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

- no relevant changes, CI should pass.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4290** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4290" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/4290?utm_source=stack-comment).

**Review by ivan-degtiarenko — COMMENTED** [review]
Left a few small comments regarding documentation.

For me, the bigger question is what is the future for `MutabilityMode`. In your implementation so far they exist in parallel - do we plan to get rid of `MutabilityMode`? Is this decision documented in the design doc or perhaps JIRA?

**Review by ivan-degtiarenko — APPROVED** [review]
@dhaus67 got it.

I think a small comment about this near the `MutabilityMode`  would be good in case someone wants to reuse it.

Also, I think you accidentally resolved https://github.com/stackrox/stackrox/pull/4290#discussion_r1064860531 even though it was not addressed.

Apart from that, everything looks good to me. Feel free to merge once you address the remaining feedback :)

File: proto/storage/traits.proto [hunk 1]
```diff
@@ -31,4 +30,21 @@ message Traits {
     HIDDEN  = 1;
   }
   Visibility visibility = 2;
+
+  // Origin specifies the origin of an object.
+  // Objects can have three different origins:
+  // - IMPERATIVE: the object was created via the API. This is assumed by default.
+  // - DEFAULT: the object is a default object, such as default image integrations, access scopes etc.
```
[the remark this answers is not in the document]
Comment (dhaus67) [reply proto/storage/traits.proto:37]: I addressed it in the wrong branch 🤦 Sorry for prematurely resolving!

**Pull request merged**